altos-rust/altos-core/src/task/control.rs

```diff
@@ -63,6 +69,13 @@ impl Priority {
     (0..NUM_PRIORITIES)
   }
 
+
+  /// Returns a box of the range of values of Priority minus the one passed in.
+  pub fn all_except<'a>(priority:Priority) -> Box<Iterator<Item=usize> + 'a> {
+    let all = Priority::all();
+    Box::new(all.filter(move |&x| x != priority as usize))
```

Comment:
I think iterators are the right choice here. They should be fairly easy to implement and if we ever need to change priority behavior we can just modify the iterator without changing the scheduling code. It'd also let us use iterators over `Priority` as opposed to `usize` in the scheduler which I think gives a better semantic feel when reading over the scheduling code.